Attraction show page

camziny opened the pull request:
add review tile.
add createReviews migration.
create Review model.
create attractionsReviews router.
create review seeder.

camziny committed:
add attraction show page.
add review tile.
add createReviews migration.
create Review model.
create attractionsReviews router.
create review seeder.

Co-authored-by: Fabricius Eusebio <yoFabricius@users.noreply.github.com>
Co-authored-by: Alex Cameron <acameron724@users.noreply.github.com>

camziny committed:
Formatting changes

camziny committed:
Merge branch 'main' into attraction-show-page

dovakeen118 commented on client/src/components/AttractionShowPage.js:
```diff
@@ -0,0 +1,52 @@
+import translateServerErrors from "../services/translateServerErrors.js";
+
+const AttractionShowPage = (props) => {
+  //debugger
```
DEBUGGER

dovakeen118 commented on client/src/components/AttractionsList.js:
```diff
@@ -1,7 +1,8 @@
 
-const AttractionsList = () => {
+const AttractionsList = (props) => {
+  //debugger
```
DEBUGGER

dovakeen118 commented on server/src/db/migrations/20220503184204_createReviews.cjs:
```diff
@@ -0,0 +1,25 @@
+        table.bigIncrements("id")
+        table.string("title").notNullable()
+        table.integer("rating").notNullable()
+        table.string("content")
```
might consider making this a type of `text` for a larger character limit on the content of the review

dovakeen118 commented on server/src/db/migrations/20220503184204_createReviews.cjs:
```diff
@@ -0,0 +1,25 @@
+ * @param {Knex} knex
+ */
+exports.up = async (knex) => {
+    return knex.schema.createTable("reviews", (table) => {
```
This file appears to have a tab size of 4 spaces, while other files in the app have a tab size of 2. This should be consistent for all group members, you can change this setting in VSCode to set the `tab size` of your editor to 2!

dovakeen118 commented on server/src/routes/api/v1/attractionsReviewsRouter.js:
```diff
@@ -0,0 +1,17 @@
+
+attractionsReviewsRouter.get("/", async (req, res) => {
+    
+    try {
```
we can remove the empty space on line 8

actually, this router isn't used at all and should be removed!

dovakeen118 commented on client/src/components/AttractionShowPage.js:
```diff
@@ -0,0 +1,52 @@
+const AttractionShowPage = (props) => {
+  //debugger
+  const [attraction, setAttraction] = useState({
+    attractions: [],
```
I don't think we want an array of `attractions` here on the show page, but instead the one attraction object with it's reviews

dovakeen118 reviewed (dismissed):
Nice work! Some debuggers to remove, a router file that isn't being used, and a couple small things to refactor

camziny committed:
removed debugger from AttractionShowPage,
removed debugger from AttractionsList,
changed content from "string" to "text" in reviews migration,
adjusted formatting in reviews migration,
removed attractionsReviewsRouter,
adjusted useState in AttractionShowPage.

camziny reviewed (commented):
removed debugger from AttractionsList,
changed content from "string" to "text" in reviews migration,
adjusted formatting in reviews migration,
removed attractionsReviewsRouter,
adjusted useState in AttractionShowPage.

nwalberts commented on client/src/components/postReview.js:
```diff
@@ -0,0 +1,25 @@
+      }
+    } catch (error) {
+      console.error(`Error in fetch: ${error.message}`);
+    }
```
this isnt a components, so put it in its own "requests" or "api" folder

nwalberts commented on server/src/db/seeders/AttractionSeeder.js:
```diff
@@ -5,24 +5,24 @@ class AttractionSeeder {
-    ]
-    for(const singleAttractionData of attractionsData) {
+        description: "Jean-Free Zone",
+      },
```
I love the goofiness, but I would seed data that is slightly less jokey since employers will get confused

nwalberts commented on server/src/routes/api/v1/attractionsRouter.js:
```diff
@@ -12,4 +14,18 @@ attractionsRouter.get("/", async (req, res) => {
+  try {
+    const attraction = await Attraction.query().findById(id)
+    attraction.reviews = await attraction.$relatedQuery("reviews")
+    return res.status(200).json({ attraction })
```
this should use a serializer, but ill let it pass for now

br-ndt committed:
address Kerrin's changes
remove attractionsReviewsRouter

br-ndt reviewed (approved):
approving these changes, out of the norm established for group projects, with permission from @nwalberts
